### PR TITLE
Added citations

### PR DESCRIPTION
**Pull request by vivekmittal514**
*Issue #, if available:*

*Description of changes:* Added citations returned by knowledge bases to streamlit application. The S3 uri is converted to a S3 presigned URL to provide source document link.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: rag-solutions/contextual-chatbot-using-knowledgebase/cfn/DeployKnowledgeBase.yaml [hunk 1]
```diff
@@ -110,12 +110,14 @@ Resources:
               response = retrieveAndGenerate(query, kb_id, model_arn, sessionId)
               generated_text = response['output']['text']
               sessionId = response['sessionId']
+              citations = response['citations']
               print (generated_text)
               print (sessionId)
-              
+              print(citations)
```
Comment (mani-aiml) [inline rag-solutions/contextual-chatbot-using-knowledgebase/cfn/DeployKnowledgeBase.yaml:116]: comment the print statement.

File: rag-solutions/contextual-chatbot-using-knowledgebase/cfn/DeployKnowledgeBase.yaml [hunk 2]
```diff
@@ -71,7 +71,7 @@ Resources:
           #print (kb_id)
 
           # declare model id for calling RetrieveAndGenerate API
-          model_id = "anthropic.claude-instant-v1"
+          model_id = "anthropic.claude-v2"
```
Comment (mani-aiml) [inline rag-solutions/contextual-chatbot-using-knowledgebase/cfn/DeployKnowledgeBase.yaml:74]: Lets update to use Claude 3 Haiku

File: rag-solutions/contextual-chatbot-using-knowledgebase/lambda/bedrock-kb-retrieveAndGenerate.py [hunk 3]
```diff
@@ -10,15 +9,15 @@
 
 # get knowledge base id from environment variable
 kb_id = os.environ.get("KNOWLEDGE_BASE_ID")
+#print (kb_id)
 
 # declare model id for calling RetrieveAndGenerate API
-model_id = "anthropic.claude-instant-v1"
+model_id = "anthropic.claude-v2"
```
Comment (mani-aiml) [inline rag-solutions/contextual-chatbot-using-knowledgebase/lambda/bedrock-kb-retrieveAndGenerate.py:15]: Lets update to use Claude 3 Haiku

**Review by mani-aiml — CHANGES_REQUESTED** [review]
Reviewed and requested changes. Please update screenshot on README showing citations, make sure to redact the s3 bucket name.

File: rag-solutions/contextual-chatbot-using-knowledgebase/cfn/DeployKnowledgeBase.yaml (hunk 2 above)
Comment (vivekmittal514) [reply rag-solutions/contextual-chatbot-using-knowledgebase/cfn/DeployKnowledgeBase.yaml]: updated to sonnet

File: rag-solutions/contextual-chatbot-using-knowledgebase/cfn/DeployKnowledgeBase.yaml (hunk 1 above)
Comment (vivekmittal514) [reply rag-solutions/contextual-chatbot-using-knowledgebase/cfn/DeployKnowledgeBase.yaml]: removed the print statement

**Pull request merged**